### DISCUSSION TITLE
Hide input based on another input's value on mount

### DISCUSSION THREAD

**Discussion by WiXSL** [react-hook-form/react-hook-form#9372] [Q&A] [answered]
Hi, I'm a [react-admin](https://github.com/marmelab/react-admin) core team member,
We are working in a component, and we have the following scenario:

In the following code sandbox, I'm trying to get an input to render hidden initially based on the value of another input: https://codesandbox.io/s/eager-hill-o4g71q?file=/src/App.tsx

The problem is that I'm NOT getting updated form values from `useWatch` until after the first render, but I can get them from `useFormContext.getValues()`,
How can I improve on this just by using `useWatch` ?

Notice that if you invert the order of the two inputs, it works as expected.

**Comment by bluebill1049** [comment]
~~This is the reason is `useWatch` need to execute before the subscription kicks in, I can take a look at this to see if we can optimise this. You want to submit a feature reqquest on this?~~

**Reply by bluebill1049** [reply]
Spend sometimes on this, it's not an easy and simple implementation as `useWatch/watch` was made for subscribe for input change and update instead of input mount, while `getValues` did get you the latest value because it's not subscribe to changes, the send re-render out the latest value.

**Comment by bluebill1049** [comment, marked as answer]
have been thinking about the issue for sometimes, we don't want to include extra computation logic into `useWatch` to keep it simple and light for subscribing to value updates. users can create a simple hook to over the issue above:

```tsx
const useFormValues = () => {
  const { getValues } = useFormContext();

  return {
    ...useWatch(),
    ...getValues(),
  }
}
```

**Reply by WiXSL** [reply]
I would really like to thank you for taking the time to dig into this.
We decided to use both `getValues` and `useWatch` as you have suggested, and it worked well.
This was just to fix a first render bug, the component was there already. If you are interested, this is the PR with the discussion about it: https://github.com/marmelab/react-admin/pull/8340
